src/core/storage/types.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Configuration options for the storage client
+ */
+export interface StorageConfig {
+  /** Private key for w3up-client authentication */
+  privateKey: string | undefined;
+  /** Delegation for storage access */
+  delegation: string | undefined;
+  /** Optional gateway URL for file retrieval */
+  gatewayUrl?: string;
+}
+
+/**
+ * File to upload
+ */
+export interface UploadFile {
+  /** Name of the file */
+  name: string;
+  /** Content of the file (base64 encoded) */
+  content: string;
+  /** MIME type of the file */
+  type?: string;
+  /** Whether to publish the file to IPFS (default: false) */
+  publishToIPFS?: boolean;
```

Comment:
This should be a per upload option not a per file option.

Reply:
yep. I removed it from here and put it in the UploadOptions only.